docs: RFC 003: Cross Language Account, Module, Message Model

aaronc opened the pull request:
# Description

[Rendered](https://github.com/cosmos/cosmos-sdk/blob/aaronc/rfc-crosslang/docs/rfc/rfc-003-crosslang.md)

This RFC conceptually specifies a framework for cross-language, cross-VM accounts and modules.

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title, you can find examples of the prefixes below:
    <!-- * `feat`: A new feature
    * `fix`: A bug fix
    * `docs`: Documentation only changes
    * `style`: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
    * `refactor`: A code change that neither fixes a bug nor adds a feature
    * `perf`: A code change that improves performance
    * `test`: Adding missing tests or correcting existing tests
    * `build`: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
    * `ci`: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
    * `chore`: Other changes that don't modify src or test files
    * `revert`: Reverts a previous commit -->
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

Please see [Pull Request Reviewer section in the contributing guide](../CONTRIBUTING.md#reviewer) for more information on how to review a pull request.

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Introduced "RFC 003: Account/Module Execution Model," outlining a framework for interoperability among accounts and modules in a multi-language environment within the Cosmos SDK.
	- Detailed the lifecycle of accounts, message transmission specifications, and the role of account handlers.

- **Documentation**
	- Added comprehensive guidelines on the execution model, including definitions, message types, and metadata requirements.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

julienrbrt reviewed (approved):
ACK.

Looking forward to hear how will that `Message data` will be defined however.
Could we mention x/account somewhere as well and how this replicates some of its features?

aaronc commented:
> Looking forward to hear how will that `Message data` will be defined however. Could we mention x/account somewhere as well and how this replicates some of its features?

I added references to x/accounts and other previous design documents.

I also added consequences and a decision section, which basically says we intend to move forward with this approach (which is my read on the discussion).

Planning to work on an update to this RFC with message data specs after this first draft is merged.

julienrbrt commented:
> Anybody know what's up with the markdown link checker? The report is long and any errors are non obvious 

It hasn't been introduced here for sure. I think it was broken for a while. Just broken links basically.
